pkg/scalers/aws_cloudwatch_scaler.go

```diff
@@ -66,9 +67,32 @@ func NewAwsCloudwatchScaler(config *ScalerConfig) (Scaler, error) {
 
 func parseAwsCloudwatchMetadata(config *ScalerConfig) (*awsCloudwatchMetadata, error) {
 	meta := awsCloudwatchMetadata{}
-	meta.metricCollectionTime = defaultMetricCollectionTime
-	meta.metricStat = defaultMetricStat
-	meta.metricStatPeriod = defaultMetricStatPeriod
+
+	if val, ok := config.TriggerMetadata["metricCollectionTime"]; ok && val != "" {
+		if n, ok := strconv.ParseInt(val, 10, 64); ok == nil {
+			meta.metricCollectionTime = n
+		} else {
+			return nil, fmt.Errorf("metricCollectionTime not a valid number")
+		}
+	} else {
+		meta.metricCollectionTime = defaultMetricCollectionTime
+	}
+
+	if val, ok := config.TriggerMetadata["metricStatPeriod"]; ok && val != "" {
+		if n, ok := strconv.ParseInt(val, 10, 64); ok == nil {
+			meta.metricStatPeriod = n
+		} else {
+			return nil, fmt.Errorf("metricStatPeriod not a valid number")
+		}
+	} else {
+		meta.metricStatPeriod = defaultMetricStatPeriod
+	}
+
+	if val, ok := config.TriggerMetadata["metricStat"]; ok && val != "" {
+		meta.metricStat = val
+	} else {
+		meta.metricStat = defaultMetricStat
+	}
```

Comment:
I think this is what the static check is complaining about with `cyclomatic complexity 35 of func `parseAwsCloudwatchMetadata` is high (> 30)`

I personally think it's fine, but maybe if you change it to

```go
	meta.metricCollectionTime = defaultMetricCollectionTime
	meta.metricStatPeriod = defaultMetricStatPeriod
	meta.metricStat = defaultMetricStat

	if val, ok := config.TriggerMetadata["metricCollectionTime"]; ok && val != "" {
		n, err := strconv.ParseInt(val, 10, 64)
		if err != nil {
			return nil, fmt.Errorf("metricCollectionTime not a valid number")
		}
		meta.metricCollectionTime = n
	}

	if val, ok := config.TriggerMetadata["metricStatPeriod"]; ok && val != "" {
		n, err := strconv.ParseInt(val, 10, 64);
		if err != nil {
			return nil, fmt.Errorf("metricStatPeriod not a valid number")
		} 
		meta.metricStatPeriod = n
	}

	if val, ok := config.TriggerMetadata["metricStat"]; ok && val != "" {
		meta.metricStat = val
	}
```

That should be equivalent to what you have but should reduce the cyclomatic complexity.

Reply:
We will try make the fill of 
`
        meta.metricCollectionTime = defaultMetricCollectionTime
	meta.metricStatPeriod = defaultMetricStatPeriod
	meta.metricStat = defaultMetricStat
`
In a function, maybe this avoid the problem with cyclomatic complexity, what you think @ahmelsayed